Library/Homebrew/search.rb

```diff
@@ -114,6 +114,61 @@ def search_formulae(string_or_regex)
     def search_casks(_string_or_regex)
       []
     end
+
+    def search_names(query, string_or_regex, args)
+      remote_results = search_taps(query, silent: true)
+
+      local_formulae = search_formulae(string_or_regex)
+      remote_formulae = remote_results[:formulae]
+      all_formulae = local_formulae + remote_formulae
+
+      local_casks = search_casks(string_or_regex)
+      remote_casks = remote_results[:casks]
+      all_casks = local_casks + remote_casks
+
+      print_formulae = args.formula?
+      print_casks = args.cask?
+      print_formulae = print_casks = true if !print_formulae && !print_casks
+      print_formulae &&= all_formulae.any?
+      print_casks &&= all_casks.any?
+
+      count = 0
+      if print_formulae
+        if $stdout.tty?
+          ohai "Formulae", Formatter.columns(all_formulae)
+        else
+          puts all_formulae
+        end
+        count += all_formulae.count
+      end
+      puts if print_formulae && print_casks
+      if print_casks
+        if $stdout.tty?
+          ohai "Casks", Formatter.columns(all_casks)
+        else
+          puts all_casks
+        end
+        count += all_casks.count
+      end
+
+      print_missing_formula_help(query, count.positive?) if local_casks.exclude?(query)
+
+      odie "No formulae or casks found for #{query.inspect}." if count.zero?
+      !count.zero?
```

Comment:
This was changed to return a boolean based upon whether any formulae or casks were found.

Reply:
```suggestion
      true
```

would work, right, given that you're doing `odie` above? Feels a bit weird to even check the return value given it'll always be `true`?

Reply:
It'll be false if no search results are returned for the search term.

Reply:
I removed it during the refactor.